Fix metadata.json file

aishwaryabh opened the pull request:
<!-- Please provide all the information below.  -->

### Issue describing the changes in this PR
The `metadata.json` file was not being generated correctly due to a bug in the script for parsing the visual studio files to have a `.zip` ending in the name, which is not valid anymore. This PR fixes the bug by updating the regular expression to match the correct file name.

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

kshyju commented:
>which is not valid anymore

Can you remind me why this is not valid anymore? since when? May be an example of the file name?

aishwaryabh commented:
> > which is not valid anymore
> 
> Can you remind me why this is not valid anymore? since when? May be an example of the file name?

Yeah so in [this PR](https://github.com/Azure/azure-functions-core-tools/pull/4201), I added a step in the consolidated pipeline where only zip up the VS artifacts after the testing has been done. Therefore, when this step is run, which is before the zipping up of the files, the script fails to find a file that ends with `.zip`. I just removed that filter for finding a file that ends with `.zip` to account for this change.
